### PR TITLE
⬆️ Upgrade Starlette to 0.17.1

### PR DESCRIPTION
**Pull request by simondale00**
⬆️ Upgrade Starlette to 0.17.0

0.17.0 addresses issue [Starlette issue #1255](https://github.com/encode/starlette/issues/1255), preventing `BaseHTTPMiddleware` from raising `anyio.ExceptionGroup`.

### CODE REVIEW

**Comment by nkamali** [comment]
Looks good to me and I'm looking forward to this upgrade for some time. Thank you, @simondale00 for helping with pushing this one forward.

**Review by nkamali — APPROVED** [review]
lgtm! 👍

**Comment by simondale00** [comment]
Hey @tiangolo! Curious what additional steps I'll need to take to help get this merged. Thanks in advance!

**Comment by Kludex** [comment]
Just wait.

**Comment by kevr** [comment]
> Just wait.

Wait for what?

It's a bit hard to decide what to do at this point.

1. An MR is up. Wait for the MR to be merged and deal with manual edits to debug things until then.
2. Depend on a fork of our own or a third party pinned commit.

If the MR is gonna take ten months, then we might as well fork and depend on that. If not, then we would just wait for the new release.

Can we get some type of... idea about the state of this and some sort of rough timeline?

**Comment by Kludex** [comment]
No one can give you an estimation, sorry.

The only one with merge rights is @tiangolo , and by the time he sees this PR, he'll probably just merge it.

If you don't want to fork, you can also pin a version predecessor to 0.69 (I'm on my phone, but I think it was 0.68.2).

**Comment by robertlagrant** [comment]
Is it possible to provide a range of versions of Starlette, or is 0.17.0 the absolute minimum?

**Comment by simondale00** [comment]
> Is it possible to provide a range of versions of Starlette, or is 0.17.0 the absolute minimum?

0.17.0 is the absolutely minimum to apply this _specific fix_, preventing Starlette from raising `anyio.ExceptionGroup`.

Though to your point, provided that Starlette adheres to semver, it might be more appropriate to pin FastAPI to >=0.17.0,<1.0.0

**Comment by Kludex** [comment]
@simondale00 [semver only kicks in after 1.0](https://semver.org/#spec-item-4). That's basically why is pinned here.

@adriangb proposed to have a pipeline that runs periodically against the latest Starlette version (or master), and would auto-merge the pin/range. Maybe we can do something about this, but it's up to @tiangolo . :kissing:

**Comment by adriangb** [comment]
> @adriangb proposed to have a pipeline that runs periodically against the latest Starlette version (or master), and would auto-merge the pin/range

Auto-bumping would be cool!

But I was thinking something much more basic: just running the pipeline and watching it for errors.
That way we can fix smaller issues (instead of fixing the accumulated issues since the last bump), and we'd also have the opportunity to make changes in Starlette before they cut a release.

**Comment by LarsStegman** [comment]
Version 0.17.1 of Starlette has been released, which fixes a bug in the authentication mechanism. I have run into this bug during development. Please use this version instead of 0.17.0.

**Comment by yezz123** [comment]
I guess there is no problem if everything works smoothly, right now the 0.16.0 do all jobs so waiting for Sebastian to bump the to the last version of starlette 🌟

**Comment by LarsStegman** [comment]
@simondale00 Could you update the PR to use 0.17.1?

**Comment by Kludex** [comment]
It really doesn't matter. @tiangolo will see the PR, and it will update it himself. Just wait for him to check this. :kissing:

**Comment by narolski** [comment]
@tiangolo I do see these exceptions raised a lot during our performance tests. Would you mind taking a look at this?

**Comment by kevr** [comment]
This time delay is kind of crazy.

File: pyproject.toml
Comment (dimaqq) [inline pyproject.toml:36]:
```suggestion
    "starlette ==0.17.1,
```

**Review by dimaqq — APPROVED** [review]

**Comment by sunshine-joe** [comment]
I also got bug with starlette<0.17.0。
wating for this PR  be merged  , thanks

**Comment by jerber** [comment]
@tiangolo this causes a lot of errors on our server, would be great if you could merge!

**Comment by narolski** [comment]
@tiangolo sorry for bugging you again, but only today I have seen this exception raised over 100 times on my pre-prod environments. Is there any way I could help with upgrading the fastapi's dependencies?

**Comment by hidaris** [comment]
> @tiangolo sorry for bugging you again, but only today I have seen this exception raised over 100 times on my pre-prod environments. Is there any way I could help with upgrading the fastapi's dependencies?

Maybe you can contact him on twitter, if the problem is really urgent

**Comment by kevr** [comment]
> Maybe you can contact him on twitter, if the problem is really urgent

It's been urgent for about a month and a half now; check out the backlog of the issue.

**Review by narolski — APPROVED** [review]

**Comment by narolski** [comment]
> > @tiangolo sorry for bugging you again, but only today I have seen this exception raised over 100 times on my pre-prod environments. Is there any way I could help with upgrading the fastapi's dependencies?
> 
> Maybe you can contact him on twitter, if the problem is really urgent

I've reached out via DM on Twitter, will let you know here should I get a response. Is there anyone else who can help out in the meantime?

**Comment by samuelcolvin** [comment]
I'd love this too, I've just pushed an empty commit to kick off tests.

**Review by samuelcolvin — APPROVED** [review]
I've tested this on a big project and everything seems to be working great

installed with
```
pip install git+https://github.com/tiangolo/fastapi.git@refs/pull/4145/merge
```

@tiangolo LGTM

**Comment by jerber** [comment]
@tiangolo could you merge this? Our production app has been getting a ton of errors a day

**Comment by dimaqq** [comment]
Looks like this project needs a backup maintainer

**Comment by dimaqq** [comment]
#4263

**Comment by tiangolo** [comment]
Thanks @simondale00! :rocket: 

And thanks a lot @Kludex and @samuelcolvin! Your help is very much appreciated. :bow: :cake: :sunglasses:

**Comment by tiangolo** [comment]
This is now available in FastAPI `0.71.0`. 🔖🚀

**Comment by samuelcolvin** [comment]
Thanks so much @tiangolo.

**Comment by Redict** [comment]
@tiangolo when it will be uploaded to pip?

**Comment by simondale00** [comment]
> @tiangolo when it will be uploaded to pip?

@Redict It's already available on PyPI: https://pypi.org/project/fastapi/0.71.0/#history

**Comment by Ashkan-Hadadi** [comment]
https://stackoverflow.com/questions/71222144/runtimeerror-no-response-returned-in-fast-api-when-refresh-request

The problem is still alive! please refer to above link

**Comment by alex-pobeditel-2004** [comment]
Hello @Ashkan-Hadadi!  
Is this issue still bothering you?  
Right now I'm stuck on FastAPI 0.75 (released on March, 5) and issue persists there. But it's interesting for me if it was solved in next versions? Looks like FastAPI got many Starlette updates recently.

**Comment by jkinkead** [comment]
@alex-pobeditel-2004 - I'm still seeing the behavior [as described in this bug](https://github.com/tiangolo/fastapi/issues/4544) on the latest version, so I think it's still an open issue.

**Comment by kozhushman** [comment]
Im using fastapi = 0.78.0 and still facing this issue, sadly

**Comment by alex-pobeditel-2004** [comment]
Looks like another fix for it will arrive with next Starlette release:  
https://github.com/encode/starlette/issues/1634

**Comment by adam-flyr** [comment]
I'm having the same issue here. Noticed that I started to happen after I changed from axios < 0.22 to 0.22. I was using cancelToken before, and AbortController now.

**Comment by Kludex** [comment]
> any update?

? 🤔

**Comment by Ki6an** [comment]
still facing the issue in fastapi = 0.79.0

**Comment by acjh** [comment]
Discussion on the Starlette repository: https://github.com/encode/starlette/discussions/1527

**Pull request merged**
